ArmVirtPkg/ArmVirtQemu.dsc

```diff
@@ -94,6 +96,23 @@
 
   ArmMonitorLib|ArmVirtPkg/Library/ArmVirtQemuMonitorLib/ArmVirtQemuMonitorLib.inf
 
+[LibraryClasses.common.SEC, LibraryClasses.common.PEI_CORE]
+  # don't check stack cookies in SEC and PEI_CORE because we don't have exception handlers installed at this point and
+  # will triple fault and reboot if we hit a stack cookie failure
+  NULL|MdePkg/Library/StackCheckLibNull/StackCheckLibNull.inf
+
+[LibraryClasses.common.PEIM]
+  NULL|MdePkg/Library/StackCheckLib/StackCheckLib.inf
+
+[LibraryClasses.AARCH64.DXE_CORE, LibraryClasses.AARCH64.DXE_DRIVER, LibraryClasses.AARCH64.DXE_RUNTIME_DRIVER, LibraryClasses.AARCH64.UEFI_DRIVER, LibraryClasses.AARCH64.UEFI_APPLICATION]
+  # for all other types, use the dynamic stack cookie
+  NULL|MdePkg/Library/StackCheckLib/StackCheckLib.inf
+  StackCheckEntryPointLib|MdePkg/Library/StackCheckLib/StackCheckLibDynamicInit.inf
+
+[LibraryClasses.ARM.DXE_CORE, LibraryClasses.ARM.DXE_DRIVER, LibraryClasses.ARM.DXE_RUNTIME_DRIVER, LibraryClasses.ARM.UEFI_DRIVER, LibraryClasses.ARM.UEFI_APPLICATION]
+  # ARM32 doesn't support dynamic stack cookies
+  NULL|MdePkg/Library/StackCheckLib/StackCheckLib.inf
+

```

Comment:
Sorry, didn't notice the last time around, but ArmVirtPkg has a `ArmVirt.dsc.inc` include file where alot of things common to all build variants (including LibraryClasses) are defined.  So I think this should be added there too.

Reply:
I can move this there, I did it here to only apply to ArmVirtQemu to start, but if you want this to apply to all the flavors, that's good with me